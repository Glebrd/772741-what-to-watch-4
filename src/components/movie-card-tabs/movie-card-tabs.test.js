import React from "react";
import renderer from "react-test-renderer";
import MovieCardTabs from "./movie-card-tabs";

const movie = {
  id: 201,
  title: `Django Unchained`,
  picture: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  genre: `Drama`,
  date: `2012`,
  poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  background: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  rating: 8.6,
  scores: 952,
  director: `Floyd Emblow`,
  starring: [`Sally Cullin`, `Harman Kleinstein`, `Chilton Petican`, `Chico Magwood`],
  description: `With the help of a German bounty hunter, a freed slave sets out to rescue his wife from a brutal Mississippi plantation owner.`,
  videoPreview: `https: //upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  runTime: `2h 45m`,
  comments: [{
    id: 1,
    user: {
      id: 356,
      name: `Oran Browning`,
    },
    rating: 10,
    comment: ` Django Unchained is superb from start to finish, it's a 2 hour and 45 minute movie yet you're on the edge of your seat, eyes glued to the screen for the whole ride. It's the closest thing to a flawless movie I have ever seen.`,
    date: `March 08, 2015`
  }, {
    id: 2,
    user: {
      id: 192,
      name: `Arman Nast`,
    },
    rating: 8,
    comment: ` Brutally hilarious and quite messy, but a total blast from start to finish`,
    date: `April 20, 2018`
  }, {
    id: 3,
    user: {
      id: 92,
      name: `Gelya Cartmale`,
    },
    rating: 10,
    comment: ` I rarely bother to give reviews after watching a movie. But holy crap this was a good movie. I'm pretty sure it is the best movie I've seen all year.`,
    date: `November 25, 2015`
  }]
};

it(`Movie card tabs component renders`, () => {
  const tree = renderer
    .create(
        <MovieCardTabs
          movie = {movie}
        />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
