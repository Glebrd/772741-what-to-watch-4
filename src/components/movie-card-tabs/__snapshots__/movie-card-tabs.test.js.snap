// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie card tabs component renders 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      .
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#!"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#!"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item "
      >
        <a
          className="movie-nav__link"
          href="#!"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      8.6
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Very good
      </span>
      <span
        className="movie-rating__count"
      >
        952
         ratings
      </span>
    </p>
  </div>
  <div
    className="movie-card__text"
  >
    <p>
      With the help of a German bounty hunter, a freed slave sets out to rescue his wife from a brutal Mississippi plantation owner.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        Floyd Emblow
         
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Sally Cullin, Harman Kleinstein, Chilton Petican, Chico Magwood
         
      </strong>
    </p>
  </div>
</div>
`;
